File: CS/VM-DrivenWizard/ViewModels/CongratulationsPageViewModel.cs
using System.ComponentModel;
using DevExpress.Mvvm;

namespace VM_DrivenWizard.ViewModels {
    public class CongratulationsPageViewModel : WizardViewModelBase, ISupportWizardFinishCommand {
        IMessageBoxService MessageBoxService => this.GetService<IMessageBoxService>();
        public CongratulationsPageViewModel() {
            ShowBack = true;
            ShowCancel = true;
            ShowFinish = true;
            AllowBack = true;
        }
        public bool CanFinish {
            get { return true; }
        }
        public void OnFinish(CancelEventArgs e) {
            MessageBoxService.ShowMessage("Thank you for completing this WPF feature tour!", "WPF Tour", MessageButton.OK, MessageIcon.Exclamation);
        }
        protected override bool GetCanCancel() {
            return false;
        }
    }
}

File: VB/VM-DrivenWizard/ViewModels/CongratulationsPageViewModel.vb
Imports System.ComponentModel
Imports DevExpress.Mvvm

Namespace VM_DrivenWizard.ViewModels

    Public Class CongratulationsPageViewModel
        Inherits WizardViewModelBase
        Implements ISupportWizardFinishCommand

        Private ReadOnly Property MessageBoxService As IMessageBoxService
            Get
                Return GetService(Of IMessageBoxService)()
            End Get
        End Property

        Public Sub New()
            ShowBack = True
            ShowCancel = True
            ShowFinish = True
            AllowBack = True
        End Sub

        Public ReadOnly Property CanFinish As Boolean Implements ISupportWizardFinishCommand.CanFinish
            Get
                Return True
            End Get
        End Property

        Public Sub OnFinish(ByVal e As CancelEventArgs) Implements ISupportWizardFinishCommand.OnFinish
            MessageBoxService.ShowMessage("Thank you for completing this WPF feature tour!", "WPF Tour", MessageButton.OK, MessageIcon.Exclamation)
        End Sub

        Protected Overrides Function GetCanCancel() As Boolean
            Return False
        End Function
    End Class
End Namespace
